ampersand-model.js

```diff
@@ -60,10 +88,18 @@ var Model = State.extend({
         wrapError(this, options);
 
         method = this.isNew() ? 'create' : (options.patch ? 'patch' : 'update');
-        if (method === 'patch') options.attrs = attrs;
+
+        if (method === 'patch') {
+            options.attrs = transformForPatch(new this.constructor(attrs).serialize(), attrs);
+        }
         // if we're waiting we haven't actually set our attributes yet so
         // we need to do make sure we send right data
-        if (options.wait && method !== 'patch') options.attrs = assign(model.serialize(), attrs);
+        if (options.wait && method !== 'patch') {
+            var clonedModel = new this.constructor(this.getAttributes({props: true}));
+            clonedModel.set(attrs);
+
+            options.attrs = assign(model.serialize(), attrs);
```

Comment:
Missed commit here. This should be `options.attrs = clonedModel.serialize();`. Will add after fix above.